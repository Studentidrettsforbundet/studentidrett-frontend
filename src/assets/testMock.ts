// ------------------- REGIONS ---------------
import {CITY, CLUB} from '../constants';

export const singleRegion = { id: 3, name: 'Vestlandet', cities: [] };
export const regionList3 = [
    { id: 0, name: 'Midtnorge', cities: [] },
    { id: 1, name: 'Oslo/Viken', cities: [] },
    { id: 2, name: 'Nord norge', cities: [] },
];
export const regionList5 = [
    { id: 0, name: 'Midtnorge', cities: [] },
    { id: 1, name: 'Oslo/Viken', cities: [] },
    { id: 2, name: 'Nord norge', cities: [] },
    { id: 3, name: 'Vestlandet', cities: [] },
    { id: 4, name: 'Sørlandet', cities: [] },
];

// ------------------- CITIES ----------------
export const singleCity = { id: 0, name: 'Trondheim', region: 'Midtnorge', clubs: [] };
export const cityList = [
    { id: 0, name: 'Trondheim', region: 'Midtnorge', clubs: [] },
    { id: 1, name: 'Oslo', region: 'Oslo/Viken', clubs: [] },
    { id: 2, name: 'Bergen', region: 'Vestlandet', clubs: [] },
    { id: 3, name: 'Stavanger', region: 'Vestlandet', clubs: [] },
];

export const invalidCityList = [
    { id: 0, name: 'Trondheim', region: 'Midtnorge', clubs: [] },
    { id: 1, name: 'Oslo', clubs: [] },
];
export const newCity = { id: 5, name: 'Orkanger', region: 'Midtnorge', clubs: [] };

// ------------------- CLUBS -----------------
export const singleClub = {
    id: 3,
    city: 1,
    name: 'NTNUI Bowling',
    description: '',
    contact_email: 'info@ntnui.no',
    membership_fee: '200',
    register_info: 'Register at ntnui.no/register',
};
export const clubList2 = [
    {
        id: 0,
        city: 0,
        name: 'NTNUI football',
        description: '',
        contact_email: 'info@ntnui.no',
        membership_fee: '200',
        register_info: 'Register at ntnui.no/register',
    },
    {
        id: 1,
        city: 0,
        name: 'NTNUI amerikansk fotball',
        description: '',
        contact_email: 'info@ntnui.no',
        membership_fee: '200',
        register_info: 'Register at ntnui.no/register',
    },
];
export const newClub = {
    id: 0,
    city: 0,
    name: 'NTNUI football',
    description: '',
    contact_email: 'info@ntnui.no',
    membership_fee: '200',
    register_info: 'Register at ntnui.no/register',
};

// ------------------- GROUPS ----------------
export const singleGroup = {
    id: 33,
    name: 'ntnui hockey',
    description: 'A hockey group at NTNUI',
    cover_photo: 'photo',
    club: 0,
    city: 3,
    contact_email: 'hockey@ntnui.no',
    sports: [],
};

export const groupList1 = [
    {
        id: 0,
        name: 'ntnui football',
        description: 'A football group at NTNUI',
        cover_photo: 'photo',
        club: 0,
        city: 0,
        contact_email: 'football@ntnui.no',
        sports: [],
    },
];

export const groupList2 = [
    {
        id: 33,
        name: 'ntnui hockey',
        description: 'A hockey group at NTNUI',
        cover_photo: 'photo',
        club: 0,
        city: 3,
        contact_email: 'hockey@ntnui.no',
        sports: [],
    },
    {
        id: 0,
        name: 'ntnui football',
        description: 'A football group at NTNUI',
        cover_photo: 'photo',
        club: 0,
        city: 0,
        contact_email: 'football@ntnui.no',
        sports: [],
    },
];

export const newGroup = {
    id: 0,
    name: 'ntnui football',
    description: 'A football group at NTNUI',
    cover_photo: 'photo',
    club: 0,
    city: 0,
    contact_email: 'football@ntnui.no',
    sports: [],
};

// ------------------- TEAMS -----------------
export const singleTeam = {
    id: 0,
    availability: 'OP',
    cost: null,
    equipment: null,
    facebook_link: null,
    gender: 'M',
    group: 1,
    image: null,
    instagram_link: null,
    location: 1,
    long_description: 'Team A is the best',
    name: 'Team A',
    schedule: [],
    tryout_dates: [],
    webpage: null,
    season: null,
    short_description: 'Short desc',
    skill_level: 'HI',
    sport: 1,
};
export const newTeam = {
    id: 4,
    availability: 'OP',
    cost: null,
    equipment: null,
    facebook_link: null,
    gender: 'M',
    group: 1,
    image: null,
    instagram_link: null,
    location: 1,
    long_description: 'Team A is the best',
    name: 'Team D',
    schedule: [],
    tryout_dates: [],
    webpage: null,
    season: null,
    short_description: 'Short desc',
    skill_level: 'HI',
    sport: 1,
};
export const teamList2 = [
    {
        id: 0,
        availability: 'OP',
        cost: null,
        equipment: null,
        facebook_link: null,
        gender: 'M',
        group: 1,
        image: null,
        instagram_link: null,
        location: 1,
        long_description: 'Team A is the best',
        name: 'Team A',
        schedule: [],
        tryout_dates: [],
        webpage: null,
        season: null,
        short_description: 'Short desc',
        skill_level: 'HI',
        sport: 1,
    },
    {
        id: 2,
        availability: 'OP',
        cost: null,
        equipment: null,
        facebook_link: null,
        gender: 'M',
        group: 1,
        image: null,
        instagram_link: null,
        location: 1,
        long_description: 'Team A is the best',
        name: 'Team B',
        schedule: [],
        tryout_dates: [],
        webpage: null,
        season: null,
        short_description: 'Short desc',
        skill_level: 'HI',
        sport: 1,
    },
];

// ------------------- SPORTS ----------------
export const singleSport = { id: 0, name: 'Football', labels: [] };
export const newSport = { id: 6, name: 'Tennis', labels: [] };
export const sportList2 = [
    { id: 0, name: 'Football', labels: [] },
    { id: 3, name: 'American football', labels: [] },
];
export const sportList5 = [
    {
        id: 0,
        name: 'Football',
        labels: [
            {
                text: 'utendørs',
                sports: ['Fotball', 'Langrenn', 'Rumpeldunk'],
                alternatives: ['Utendørs'],
            },
        ],
    },

    {
        id: 1,
        name: 'Handball',
        labels: [
            {
                text: 'utendørs',
                sports: ['Hanball'],
                alternatives: ['Indoors'],
            },
        ],
    },
    {
        id: 2,
        name: 'Basketball',
        labels: [
            {
                text: 'utendørs',
                sports: ['Basketball'],
                alternatives: ['Utendørs'],
            },
        ],
    },
    {
        id: 3,
        name: 'American football',
        labels: [
            {
                text: 'utendørs',
                sports: ['Fotball'],
                alternatives: ['Utendørs'],
            },
        ],
    },
    {
        id: 4,
        name: 'Baseball',
        labels: [
            {
                text: 'utendørs',
                sports: ['Baseball'],
                alternatives: ['Utendørs'],
            },
        ],
    },
];

// ------------------- SEARCH ----------------
export const searchResult = { name: 'Test', id: '1', label: CITY };
export const newSearchResult = { name: 'Test2', id: '2', label: CLUB };
