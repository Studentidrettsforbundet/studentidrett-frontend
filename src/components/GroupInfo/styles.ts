import { style, classes } from 'typestyle';
import colors from '../../styles/colors';

export const coverImage = style({
    maxWidth: '100%',
    height: 'auto',
    marginRight: 'auto',
    marginLeft: 'auto',
    display: 'block'
})