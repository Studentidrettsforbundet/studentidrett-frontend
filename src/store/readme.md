# Redux

This folder contains all redux files.

Add this extension to your browser: https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd

## Description:

store.ts is the file that creates the redux store, as well as the combined reducer. All reducers needs to be added to the combined reducer, and to the combinedState

This folder contains a folder for each page of our application that needs redux.
The folders contains reducers, actions, action crators, types, interfaces, etc. related to pages.

## Example of folder structure

We have a page called "search". That page needs a reducer, some actions, action creators, and some interfaces. Below is an example of the folder structure

----/store
--------store.ts
--------/pages
-----------/search
-----------------searchReducer.ts
-----------------searchActions.ts
-----------------searchReducer.ts
