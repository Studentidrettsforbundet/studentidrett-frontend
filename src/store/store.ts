import { applyMiddleware, combineReducers, compose, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import { cityInitialState, cityReducer, cityState } from './pages/city/cityReducer';
import { clubInitialState, clubReducer, clubDetailReducer, clubState } from './pages/club/clubReducer';
import { groupInitialState, groupReducer, groupDetailReducer, groupState } from './pages/group/groupReducer';
import { regionInitialState, regionReducer, regionState } from './pages/region/regionReducer';
import { sportInitialState, sportReducer, sportDetailReducer, sportState } from './pages/sport/sportReducer';
import { teamInitialState, teamReducer, teamDetailReducer, teamState } from './pages/team/teamReducer';
import { searchBarInitialState, searchBarReducer, searchBarState } from './searchBar/searchBarReducer';
import { thunkInitialState, thunkReducer, thunkState } from './thunks/thunkReducer';
import { searchInitialState, searchReducer, searchState } from './pages/search/searchReducer';
import {
    questionInitialState,
    questionnaireReducer,
    questionnaireState,
} from './pages/questionnaire/questionnaireReducer';

declare global {
    interface Window {
        __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
    }
}

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

//TODO: add reducers interfaces
export interface combinedStateInterface {
    region: regionState;
    city: cityState;
    sport: sportState;
    sport_detail: sportState;
    club: clubState;
    club_detail: clubState;
    team: teamState;
    team_detail: teamState;
    group: groupState;
    group_detail: groupState;
    search_results: searchState;
    thunk: thunkState;
    searchBar: searchBarState;
    questionnaire: questionnaireState;
}

// TODO: add all pages state here
export const combinedState = {
    region: regionInitialState,
    city: cityInitialState,
    sport: sportInitialState,
    sport_detail: sportInitialState,
    club: clubInitialState,
    club_detail: clubInitialState,
    team: teamInitialState,
    team_detail: teamInitialState,
    group: groupInitialState,
    group_detail: groupInitialState,
    search_results: searchInitialState,
    thunk: thunkInitialState,
    searchBar: searchBarInitialState,
    questionnaire: questionInitialState,
};

// create store
const store = createStore(
    combineReducers({
        region: regionReducer,
        city: cityReducer,
        sport: sportReducer,
        sport_detail: sportDetailReducer,
        club: clubReducer,
        club_detail: clubDetailReducer,
        team: teamReducer,
        team_detail: teamDetailReducer,
        group: groupReducer,
        group_detail: groupDetailReducer,
        search_results: searchReducer,
        thunk: thunkReducer,
        searchBar: searchBarReducer,
        questionnaire: questionnaireReducer,
    }),
    combinedState,
    composeWithDevTools(applyMiddleware(thunk)),
);

export default store;
