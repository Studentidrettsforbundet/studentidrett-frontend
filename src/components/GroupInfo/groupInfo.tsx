import React from 'react';
import image from '../../assets/placeholder.png';
import { coverImage } from './styles';

interface IClubInfo {
    title: string;
    cover_photo?: string;
    description: string;
}

const GroupInfo = (props: IClubInfo): JSX.Element => {
    return (
        <React.Fragment>
            <h1>{props.title}</h1>
            <img className={coverImage} alt={props.title} src={image}/>
            <br/>
            <p>{props.description}</p> 
        </React.Fragment>
    );
};

export default GroupInfo;
