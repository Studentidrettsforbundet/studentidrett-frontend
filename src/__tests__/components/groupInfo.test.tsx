import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import GroupInfo from '../../components/GroupInfo/groupInfo';
import store from '../../store/store';
import { singleGroup } from '../../assets/testMock';

describe('clubCard', () => {
    test('renders correctly', () => {
        const tree = renderer.create(
            <Provider store={store}>
                <MemoryRouter initialEntries={['/sports/1']}>
                    <GroupInfo title={singleGroup.name} description={singleGroup.description} />
                </MemoryRouter>
            </Provider>,
        );
        expect(tree).toMatchSnapshot();
    });
});
