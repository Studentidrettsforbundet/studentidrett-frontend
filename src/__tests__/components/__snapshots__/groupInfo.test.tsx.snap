// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clubCard renders correctly 1`] = `
Array [
  <h1>
    ntnui hockey
  </h1>,
  <img
    alt="ntnui hockey"
    className="fkuizqk"
    src="placeholder.png"
  />,
  <br />,
  <p>
    A hockey group at NTNUI
  </p>,
]
`;
